### PR TITLE
refactor: Removing unnecessary build outputs

### LINKED ISSUE #1495: Build creates 200.html with `--no-prerender` option
**Do you want to request a _feature_ or report a _bug_?**
Bug

**What is the current behaviour?**
`preact build --no-prerender` creates a `200.html` file.

**If the current behaviour is a bug, please provide the steps to reproduce.**

**What is the expected behaviour?**
The file should not be created, as in v3.0.3.

**If this is a feature request, what is motivation or use case for changing the behaviour?**

**Please mention other relevant information.**

Please paste the results of `preact info` here.
```
Environment Info:
  System:
    OS: macOS 10.15.7
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
  Binaries:
    Node: 15.3.0 - ~/.n/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.0.14 - ~/.n/bin/npm
  Browsers:
    Chrome: 87.0.4280.88
    Edge: 87.0.664.57
    Firefox: 83.0
    Safari: 14.0.1
  npmPackages:
    preact: ^10.5.7 => 10.5.7
    preact-cli: ^3.0.5 => 3.0.5
```

### PR DESCRIPTION
**Pull request by rschristian**
**What kind of change does this PR introduce?**

Removing unnecessary files from being output by build

**Did you add tests for your changes?**

Edited existing

**Summary**

Closes #1495 

**Does this PR introduce a breaking change?**

No

### CODE REVIEW

File: packages/cli/tests/build.test.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ function getRegExpFromMarkup(markup) {
 function testMatch(src, tar) {
 	let k, tmp;
 	let keys = Object.keys(tar);
-	expect(keys).toHaveLength(Object.keys(src).length);
+	expect(Object.keys(src)).toHaveLength(keys.length);
```
Comment (rschristian) [inline packages/cli/tests/build.test.js:45]: This was backwards which made the test failure output rather confusing.

**Review by prateekbh — CHANGES_REQUESTED** [review]
When removing `200.html` this line needs to fallback to `index.html`
https://github.com/preactjs/preact-cli/blob/master/packages/cli/sw/index.js#L23

File: packages/cli/sw/index.js [hunk 2]
```diff
@@ -20,7 +20,10 @@ export function setupRouting() {
 
 	setCatchHandler(({ event }) => {
 		if (isNav(event)) {
-			return caches.match(getCacheKeyForURL('/200.html'));
+			return (
+				caches.match(getCacheKeyForURL('/200.html')) ||
+				caches.match(getCacheKeyForURL('/index.html'))
+			);
 		}
```
Comment (rschristian) [inline packages/cli/sw/index.js:27]: @prateekbh Sorry, missed that. Hopefully this will work?

Comment (rschristian) [reply packages/cli/sw/index.js:27]: nvm, need to at least touch up the test suite first

Comment (rschristian) [reply packages/cli/sw/index.js:27]: Alright, should be corrected now.

Comment (prateekbh) [reply packages/cli/sw/index.js:27]: sry for late reply would `caches.match(getCacheKeyForURL('/200.html') || getCacheKeyForURL('/index.html'))` work?

Comment (rschristian) [reply packages/cli/sw/index.js:27]: No worries. Should do. Not sure why I did it like that.

Comment (rschristian) [reply packages/cli/sw/index.js]: Corrected. Thanks for the catch.

**Comment by privateOmega** [comment]
@rschristian Does this PR require any more changes to be merged. Thanks for the contribution, I am new to preact, but would love to see this land.

**Comment by rschristian** [comment]
> 
> @rschristian Does this PR require any more changes to be merged? Thanks for the contribution, I am new to preact, but would love to see this land.

Looks like I need to rebase the changes but otherwise that'd be up to Prateek/anyone else who had improvements/concerns.

**Review by prateekbh — APPROVED** [review]

**Comment by prateekbh** [comment]
@rschristian as a follow up can you add a test similar to https://github.com/preactjs/preact-cli/blob/master/packages/cli/tests/service-worker.test.js#L48-L63 but with pre-rendering disabled

**Review by ForsakenHarmony — APPROVED** [review]

**Comment by rschristian** [comment]
@prateekbh Sorry, hadn't gotten the chance to add a test yet due to the holidays. Figured I'd pick it back up on Monday (probably)

**Comment by maltoze** [comment]
Waiting for release

**Pull request merged**